clients/segments/segments.go

```diff
@@ -133,44 +125,72 @@ func (c Client) Update(ctx context.Context, id string, data []byte) (Response, e
 	}
 	err = json.Unmarshal(existing.Data, &getResponse)
 	if err != nil || getResponse.Version == 0 || getResponse.Owner == "" {
-		return Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
 	}
 
 	// Adds owner if not set(they are mandatory from the API),
 	// and always override uid with the uid that is getting updated
-	data, err = addOwnerAndUIDIfNotSet(data, getResponse.Owner, id)
+	body, err = addOwnerAndUIDIfNotSet(body, getResponse.Owner, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+
+	path, err := url.JoinPath(endpointPath, id)
 	if err != nil {
-		return Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
 	}
 
-	updateResourceResp, err := c.client.Update(ctx, id, data, rest.RequestOptions{
+	resp, err := c.restClient.PUT(ctx, path, bytes.NewReader(body), rest.RequestOptions{
 		CustomShouldRetryFunc: rest.RetryIfTooManyRequests,
 		QueryParams:           map[string][]string{"optimistic-locking-version": {fmt.Sprint(getResponse.Version)}}})
```

Comment:
Would this work? I had to do a double-take at `fmt.Sprint(...)`...
```suggestion
		QueryParams:           map[string][]string{"optimistic-locking-version": {strconv.Itoa(getResponse.Version)}}})
```